sympy/physics/continuum_mechanics/beam.py

```diff
@@ -0,0 +1,472 @@
+"""
+This module can be used to solve 2D beam bending problems with
+singularity functions in mechanics.
+
+"""
+
+from __future__ import print_function, division
+
+from sympy.core import S, Symbol, diff
+from sympy.solvers import linsolve
+from sympy.printing import sstr
+from sympy.functions import SingularityFunction
+from sympy.core import sympify
+from sympy.integrals import integrate
+
+
+class Beam(object):
+    """
+    A Beam is a structural element that is capable of withstanding load
+    primarily by resisting against bending. Beams are characterized by
+    their cross sectional profile(Second moment of area), their length
+    and their material.
+
+
+    Examples
+    ========
+    There is a beam of length 4 meters. A constant distributed load of 6 Nm/m
+    is applied from half of the beam till the end. There are two supports below
+    the beam, one at the starting point and another at the ending point of the beam.
+    The deflection of the beam at the end is resticted.
+
+    >>> from sympy.physics.continuum_mechanics.beam import Beam
+    >>> from sympy import Symbol, Piecewise
+    >>> x = Symbol('x')
+    >>> E = Symbol('E')
+    >>> I = Symbol('I')
+    >>> b = Beam(4, E, I)
+    >>> b.apply_load(value=-9, start=4, order=-1)
+    >>> b.apply_load(value=-3, start=0, order=-1)
+    >>> b.apply_load(order=0, start=2, value=6)
+    >>> b.bc_deflection = [(4, 0)]
+    >>> b.boundary_conditions
+    {'deflection': [(4, 0)], 'moment': [], 'slope': []}
+    >>> b.load
+    -3*SingularityFunction(x, 0, -1) + 6*SingularityFunction(x, 2, 0) - 9*SingularityFunction(x, 4, -1)
+    >>> b.shear_force()
+    -3*SingularityFunction(x, 0, 0) + 6*SingularityFunction(x, 2, 1) - 9*SingularityFunction(x, 4, 0)
+    >>> b.bending_moment()
+    3*SingularityFunction(x, 0, 1) - 3*SingularityFunction(x, 2, 2) + 9*SingularityFunction(x, 4, 1)
+    >>> b.slope()
+    (3*SingularityFunction(x, 0, 2)/2 - SingularityFunction(x, 2, 3) + 9*SingularityFunction(x, 4, 2)/2 - 7)/(E*I)
+    >>> b.deflection()
+    (-7*x + SingularityFunction(x, 0, 3)/2 - SingularityFunction(x, 2, 4)/4 + 3*SingularityFunction(x, 4, 3)/2)/(E*I)
```

Comment:
Pretty printing is making this more bad . See here : 
```py
File "sympy/physics/continuum_mechanics/beam.py", line 45, in sympy.physics.continuum_mechanics.beam.Beam
Failed example:
    b.boundary_conditions
Expected:
    {'deflection': [(0, 0), (4, 0)], 'moment': [], 'slope': []}
Got:
    {deflection: [(0, 0), (4, 0)], moment: [], slope: []}
**********************************************************************
File "sympy/physics/continuum_mechanics/beam.py", line 47, in sympy.physics.continuum_mechanics.beam.Beam
Failed example:
    b.load
Expected:
    -3*SingularityFunction(x, 0, -1) + 6*SingularityFunction(x, 2, 0) - 9*SingularityFunction(x, 4, -1)
Got:
    -3*SingularityFunction(x, 0, -1) + 6*SingularityFunction(x, 2, 0) - 9*Singular
    ityFunction(x, 4, -1)
**********************************************************************
File "sympy/physics/continuum_mechanics/beam.py", line 49, in sympy.physics.continuum_mechanics.beam.Beam
Failed example:
    b.shear_force()
Expected:
    -3*SingularityFunction(x, 0, 0) + 6*SingularityFunction(x, 2, 1) - 9*SingularityFunction(x, 4, 0)
Got:
    -3*SingularityFunction(x, 0, 0) + 6*SingularityFunction(x, 2, 1) - 9*Singulari
    tyFunction(x, 4, 0)
**********************************************************************
File "sympy/physics/continuum_mechanics/beam.py", line 51, in sympy.physics.continuum_mechanics.beam.Beam
Failed example:
    b.bending_moment()
Expected:
    3*SingularityFunction(x, 0, 1) - 3*SingularityFunction(x, 2, 2) + 9*SingularityFunction(x, 4, 1)
Got:
    3*SingularityFunction(x, 0, 1) - 3*SingularityFunction(x, 2, 2) + 9*Singularit
    yFunction(x, 4, 1)
**********************************************************************
File "sympy/physics/continuum_mechanics/beam.py", line 53, in sympy.physics.continuum_mechanics.beam.Beam
Failed example:
    b.slope()
Expected:
    (3*SingularityFunction(x, 0, 2)/2 - SingularityFunction(x, 2, 3) + 9*SingularityFunction(x, 4, 2)/2 - 7)/(E*I)
Got:
    3*SingularityFunction(x, 0, 2)                                  9*SingularityF
    ------------------------------ - SingularityFunction(x, 2, 3) + --------------
                  2                                                               
    ------------------------------------------------------------------------------
                                                   E*I                            
    <BLANKLINE>
    unction(x, 4, 2)    
    ---------------- - 7
    2                   
    --------------------
    <BLANKLINE>
**********************************************************************
File "sympy/physics/continuum_mechanics/beam.py", line 55, in sympy.physics.continuum_mechanics.beam.Beam
Failed example:
    b.deflection()
Expected:
    (-7*x + SingularityFunction(x, 0, 3)/2 - SingularityFunction(x, 2, 4)/4 + 3*SingularityFunction(x, 4, 3)/2)/(E*I)
Got:
           SingularityFunction(x, 0, 3)   SingularityFunction(x, 2, 4)   3*Singula
    -7*x + ---------------------------- - ---------------------------- + ---------
                        2                              4                          
    ------------------------------------------------------------------------------
                                                    E*I                           
    <BLANKLINE>
    rityFunction(x, 4, 3)
    ---------------------
         2               
    ---------------------
    <BLANKLINE>
**********************************************************************
File "sympy/physics/continuum_mechanics/beam.py", line 57, in sympy.physics.continuum_mechanics.beam.Beam
Failed example:
    b.deflection().rewrite(Piecewise)
Expected:
    (-7*x + Piecewise((x**3, x > 0), (0, True))/2
          + 3*Piecewise(((x - 4)**3, x - 4 > 0), (0, True))/2
          - Piecewise(((x - 2)**4, x - 2 > 0), (0, True))/4)/(E*I)
Got:
           / 3                //       3               \   /       4              
           |x   for x > 0     ||(x - 4)   for x - 4 > 0|   |(x - 2)   for x - 2 > 
           <                3*|<                       |   <                      
           |0   otherwise     ||   0        otherwise  |   |   0        otherwise 
           \                  \\                       /   \                      
    -7*x + -------------- + ---------------------------- - -----------------------
                 2                       2                            4           
    ------------------------------------------------------------------------------
                                          E*I                                     
    <BLANKLINE>
    <BLANKLINE>
    0
    <BLANKLINE>
    <BLANKLINE>
    <BLANKLINE>
    -
    <BLANKLINE>
    -
    <BLANKLINE>

```